### ISSUE TITLE
Support for large field viewing, E. g. JSON

### ISSUE THREAD

**Issue by mottykohn** [microsoft/azuredatastudio#313]
- SQL Operations Studio Version:
0.23
Steps to Reproduce:
Query any large field

We started taking advantage of the new JSON support in SQL server 2016. Problem is that in one field there could be a lot of data and there is no good way to view it aside for saving it to a file or copy and paste to a text editor. It would be nice if there would be a built in viewer for JSON. thanks.

**Comment by sqlruss** [comment]
It does contain a built-in JSON viewer, at least as of 0.25.4. Result-sets returned with the FOR JSON TSQL clause can be clicked to open the JSON document in a new tab under the JSON language mode. However, a result-set that includes an NVARCHAR attribute already formatted as JSON is not recognized as JSON.

**Comment by mottykohn** [comment]
So can it be added something like when you right click there is an option to view as JSON?

Maybe someone can come up with a better design idea but the point is that it should be view-able without copy and paste.

**Comment by sqlruss** [comment]
If you querying a SQL Server 2016 instance, you can transform any result-set output into JSON with the FOR JSON clause, regardless of the database compatibility level. SQL Operations Studio recognizes the JSON in the result-set and highlights it as though it is a hyperlink. Clicking on it will cause it to automatically open it in a new window with JSON formatting. However, this does not occur if the data is actually stored as JSON in a SQL 2016 NVARCHAR, which is the missing use case.

**Comment by mottykohn** [comment]
I got that, I mean the latter where I'm storing a long JSON string in a [N]VARCHAR. I store REST PUTs as is in my table then an ETL transforms it. When I want to view the input tables while debugging as it stands now is a pain.

**Comment by kevcunnane** [comment]
This would be a nice extensibility point - as someone mentioned right-click and have options to call through and retrieve cell data (then display in new window) could be a cool sample extension.

**Comment by sqlruss** [comment]
Ah. Returning to this a month later, I've just realized that there is a way to return data already stored as JSON such that SQL Operations Studio will automatically format it as JSON. I had missed this use case for the JSON_QUERY function. "If you're returning results with FOR JSON, and you're including data that's already in JSON format (in a column or as the result of an expression), wrap the JSON data with JSON_QUERY without the path parameter." https://docs.microsoft.com/en-us/sql/t-sql/functions/json-query-transact-sql

**Comment by alanrenmsft** [comment]
This has been fixed, we do a check on all the cells and if it is JSON, you can click on the cell and the content will be opened in a separate editor as JSON.

**Comment by mottykohn** [comment]
Thanks @alanrenmsft!
Just one question, and let me know if I should open a new case for this
if JSON contains an array this does not work
```
SELECT '{ "one": [1,2] }'
UNION
SELECT '{ "two": 2 }'
```
note that row "one" will not be clickable

**Comment by alanrenmsft** [comment]
@mottykohn what version are you using?

Lastest version works fine for me: 
![image](https://user-images.githubusercontent.com/13777222/206576090-d2172522-2a6f-4a9d-aaef-51687c62b3d3.png)

**Comment by mottykohn** [comment]
I see my problem. I tried it on my production data and was not working, so I thought it had something to do with arrays, but now I figured it out. My JSON fields are formatted and that is what is giving me problems.
![image](https://user-images.githubusercontent.com/17966568/206585430-31d4cc32-09ee-4cb9-bdee-9340f304e1b1.png)

Version: 1.40.1 (system setup)
Commit: 3a32679ab590c1023ab4de6f191769e7e127d25b
Date: 2022-11-22T18:30:25.850Z
VS Code: 1.67.0
Electron: 17.4.11
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.14-electron.0
OS: Windows_NT x64 10.0.22000
